azurerm/resource_arm_network_interface_card.go

```diff
@@ -285,29 +288,49 @@ func resourceArmNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) e
 		}
 	}
 
+	if iface.IPConfigurations != nil {
+		d.Set("ip_configuration", schema.NewSet(resourceArmNetworkInterfaceIpConfigurationHash, flattenNetworkInterfaceIPConfigurations(iface.IPConfigurations)))
+	}
+
 	if iface.VirtualMachine != nil {
 		if *iface.VirtualMachine.ID != "" {
 			d.Set("virtual_machine_id", *iface.VirtualMachine.ID)
 		}
 	}
 
+	var appliedDNSServers []string
+	var dnsServers []string
 	if iface.DNSSettings != nil {
 		if iface.DNSSettings.AppliedDNSServers != nil && len(*iface.DNSSettings.AppliedDNSServers) > 0 {
-			dnsServers := make([]string, 0, len(*iface.DNSSettings.AppliedDNSServers))
-			for _, dns := range *iface.DNSSettings.AppliedDNSServers {
-				dnsServers = append(dnsServers, dns)
+			for _, applied := range *iface.DNSSettings.AppliedDNSServers {
+				appliedDNSServers = append(appliedDNSServers, applied)
 			}
+		}
 
-			if err := d.Set("applied_dns_servers", dnsServers); err != nil {
-				return err
+		if iface.DNSSettings.DNSServers != nil && len(*iface.DNSSettings.DNSServers) > 0 {
+			for _, dns := range *iface.DNSSettings.DNSServers {
+				dnsServers = append(dnsServers, dns)
 			}
 		}
 
 		if iface.DNSSettings.InternalFqdn != nil && *iface.DNSSettings.InternalFqdn != "" {
 			d.Set("internal_fqdn", iface.DNSSettings.InternalFqdn)
 		}
+
+		d.Set("internal_dns_name_label", iface.DNSSettings.InternalDNSNameLabel)
+	}
+
+	if iface.NetworkSecurityGroup != nil {
+		d.Set("network_security_group_id", resp.NetworkSecurityGroup.ID)
 	}
 
+	d.Set("applied_dns_servers", appliedDNSServers)
+	d.Set("dns_servers", dnsServers)
+	d.Set("enable_ip_forwarding", resp.EnableIPForwarding)
+	d.Set("location", resp.Location)
```

Comment:
_minor_: it'd be good to use:
```
d.Set("location", azureRMNormalizeLocation(*resp.Location))
``` 

here which ensures consistency in the statefile - I'll push this change in a follow up PR